File: src/differentiation/C/explicit_euler.c
#include "functions.h"
#include <stdio.h>

double explicit_euler(double (*f)(double, double, double), double t, double y, double alpha, double beta, char v)
{
    int n = alpha / beta;

    if (v)
    {
        printf("%12s%12s%12s\n", "STEP (k)", "t_k", "y_k");
    }

    for (int k = 0; k < n; k++)
    {
        if (v)
        {
            printf("%12d%12f%12f\n", k, t, y);
        }
        y += f(y, alpha, beta) * beta;
        t += beta;
    }

    return y;
}

File: src/differentiation/explicit_euler.py
import numpy as np

def explicit_euler(f, T, f0, step, n, verbose=False):
    Y = np.zeros(n)
    Y[0] = f0
    if verbose:
        print("%12s%12s%12s".format("step (k)", "y", "t"))
    y = Y[0]
    for k in range(n - 1):
        y += step * f(T[k], Y[k])
        Y[k + 1] = y
        if verbose:
            print("%12d%12f%12f".format(k, Y[k], T[k]))

    return Y